energy_hub/ehub_model.py

```
@@ -503,10 +425,22 @@ def _add_indexed_constraints(self):
 
             logging.info(f'{rule.__name__} is enabled')
 
-            name = rule.__name__ + '_constraint'
-            args = [getattr(self._model, arg) for arg in rule.args]
+            if not rule.args:
+                self.constraints.append(rule())
+            else:
+                sets = [getattr(self, arg) for arg in rule.args]
+
+                for arguments in product(*sets):
+                    expression = rule(*arguments)
 
-            setattr(self._model, name, Constraint(*args, rule=rule))
+                    # Comparing to True and False is bad, but expression can
+                    # be many types
+                    if expression is True:
+                        continue
+                    elif expression is False:
+                        raise RuntimeError
```

Might want to raise a better exception.